### ISSUE TITLE
SelectArrayInput has no multiselection indication for screenreaders

### ISSUE THREAD

**Issue by tiagoit** [marmelab/react-admin#7053]
**What you were expecting:**
When using a SelectArrayInput the screenreaders don't indicate that this field can have a multiselection as we have with a regular select element containing the multiple attribute.

**What happened instead:**
Screenreaders read the element as a regular list of items.

**Steps to reproduce:**
Use a screenreader application to inspect a SelectArrayInput component.

**Related code:**
```
<ReferenceArrayInput source="serviceTypesIds" reference="service-types" fullWidth>
    <SelectArrayInput optionText="name" />
</ReferenceArrayInput>
```

**Other information:**

**Environment**

* React-admin version: 3.15.2
* Last version that did not exhibit the issue (if applicable):
* React version: 17.0.2
* Browser: 96.0.4664.110
* Stack trace (in case of a JS error):

**Comment by frastlin** [comment]
Here is an example of correct behavior, see the multiselect heading:
https://mui.com/components/selects/
Here is a description of what should happen:
http://whatsock.com/Templates/Listboxes/Multiselect/index.htm
Here is a free screen reader to test on Windows:
https://www.nvaccess.org/download/

**Comment by fzaninotto** [comment]
This was a MUI bug (https://github.com/mui/material-ui/issues/38631) fixed in September 2023 (https://github.com/mui/material-ui/pull/38855) and released in MUI v5.14.11 (https://github.com/mui/material-ui/releases/tag/v5.14.11).
